docs: refine quick start

aorumbayev opened the pull request:
Addresses #486 

## Proposed Changes

  - Refreshing quick start page
  - Patching ci, macos-latest is now arm by default, adding intel explicitly

neilcampbell commented on docs/tutorials/intro.md:
````diff
@@ -70,102 +72,72 @@ algokit init
 
 ```shell
-algokit init -t playground --no-workspace
+algokit init -t python --no-workspace -a preset_name "starter"
````
Looks like my previous update was never synced to the docs site.

neilcampbell commented on docs/tutorials/intro.md:
```diff
@@ -70,102 +72,72 @@ algokit init
 Next, you will be prompted for the name of your project. Finally, select the default value for the rest of the prompts (enter).
 
-Once finished, (if you have it installed) VS Code should automatically be opened with the initialised project and you will be prompted to install appropriate VS Code extensions. This starter app will contain one smart contract (built using the [Beaker](https://beaker.algo.xyz/) smart contract development framework) named `helloworld.py`, in the `hello_world` folder, with one method (`hello`) that ...
+Once finished, (if you have it installed) VS Code should automatically be opened with the initialised project and you will be prompted to install appropriate VS Code extensions. This starter app will contain one smart contract (built using the [Algorand Python](https://algorandfoundation.github.io/puya/) language) named `contract.py`, in the `smart_contracts/hello_world` folder, with one method ( ...
```
```suggestion
Once finished, (if you have it installed) VS Code should automatically be opened with the initialised project and you will be prompted to install appropriate VS Code extensions. This starter app will contain one smart contract (built using the [Algorand Python](https://algorandfoundation.github.io/puya/) language) named `contract.py`, in the `smart_contracts/hello_world` folder, with one method (`hello`) that takes a `String` and returns a `String`. Notice that within the contract folder, there is a `deploy_config.py` which defines the parameters for deployment of your smart contract to the target chain (Algorand TestNet, Algorand MainNet, Algorand LocalNet or else).
```

neilcampbell commented on docs/tutorials/intro.md:
```diff
@@ -70,102 +72,72 @@ algokit init
+## AlgoKit Project structure 🏗
 
-Once the playground project is created, you will notice in the `hello_world` folder a file named `demo.py` which is a simple example of using AlgoKit to deploy and make a call to the `helloworld.py` smart contract on the LocalNet instance started earlier.
+Since you have just instantiated an official AlgoKit template, each official smart contract template includes an interactive codespace walkthrough powered by [CodeTours](https://marketplace.visualstudio.com/items?itemName=vsls-contrib.codetour).
```
```suggestion
Since you have just instantiated an official AlgoKit template, each official smart contract template includes an interactive codespace walkthrough powered by [CodeTour](https://marketplace.visualstudio.com/items?itemName=vsls-contrib.codetour).
```

neilcampbell commented on docs/tutorials/intro.md:
```diff
@@ -70,102 +72,72 @@ algokit init
-1. Start LocalNet
-2. Build the smart contract
-3. Deploy and call the smart contract (`demo.py`)
+Upon execution, the walkthrough will guide you through the key components of your fresh AlgoKit project instance, which is an important prerequisite to learn before running and deploying your first smart contract. Certainly, not because its hard to learn, but because it will save you a lot of time and efforts if you set the right foundation first.
```
```suggestion
Upon execution, the walkthrough will guide you through the key components of your fresh AlgoKit project, which is an important prerequisite to learn before running and deploying your first smart contract. Certainly, not because its hard to learn, but because it will save you a lot of time and effort having the correct foundations.
```

neilcampbell commented on docs/tutorials/intro.md:
```diff
@@ -70,102 +72,72 @@ algokit init
+[![Watch the video](https://ipfs.algonode.xyz/ipfs/Qmc9mRaPoDyhUFmek4ETxVfKUKzUg9pf3Tss5xwkBGdQis)](https://ipfs.algonode.xyz/ipfs/Qma6gNqxsSFc9Jbh8kBTZyVLv5gqFj1xnrsjoeT6MAAwCw/)
 
-The App ID of of the deployed contract and its Algorand address is displayed, followed by the message returned from the smart contract call (`Hello, Beaker`).
+As shown above, click on clicking on `Run and Debug` and then `Build & Deploy contracts`. Alternatively, you can hit F5 (or whatever keyboard shortcut is set in VS Code for running the debugger). While you are viewing the `contract.py` file - it will:
```
```suggestion
As shown above, click on `Run and Debug` and then `Build & Deploy contracts`. Alternatively, you can hit F5 (or whatever keyboard shortcut is set in VS Code for running the debugger). While you are viewing the `contract.py` file - it will:
```

neilcampbell commented on docs/tutorials/intro.md:
```diff
@@ -70,102 +72,72 @@ algokit init
+3. Deploy and call the `HelloWorld` smart contract (from `contract.py` file).
 
-Additionally, you can find the native TEAL smart contract code and the appropriate smart contract manifest JSON files have been output to the `artifacts` folder.
+After execution, the `App ID` of of the deployed contract and its Algorand address are displayed, followed by the message returned from the smart contract call (`Hello, world`).
```
```suggestion
After execution, the `App ID` of the deployed contract and its Algorand address are displayed, followed by the message returned from the smart contract call (`Hello, world`).
```

neilcampbell commented on docs/tutorials/intro.md:
```diff
@@ -70,102 +72,72 @@ algokit init
+At this point you have deployed a simple contract to a LocalNet Algorand network (running in your Docker environment) and called it successfully!
 
-Running the `build.py` python file will also generate these artifacts without deploying. These files can be used by tools like [Dappflow](https://dappflow.org/), [goal](https://developer.algorand.org/docs/clis/goal/goal/), etc. to deploy your smart contract to the various Algorand networks.
+Additionally, you can find the native TEAL smart contract code and the appropriate smart contract manifest JSON files have been output to the `artifacts` folder. Note that in this example, TEAL code is compiled using `PuyaPy` compiler, which is what `Algorand Python` relies on to represent your Python smart contract code in TEAL ([Transaction Execution Approval Language](https://developer.algoran ...
```
```suggestion
Additionally, you can find the native TEAL smart contract code and the appropriate smart contract manifest JSON files have been output to the `artifacts` folder. Note that in this example, TEAL code is compiled using the `PuyaPy` compiler, which is what `Algorand Python` relies on to compile your Python smart contract code to TEAL ([Transaction Execution Approval Language](https://developer.algorand.org/docs/get-details/dapps/avm/teal/)).
```

aorumbayev replied:
@neilcampbell remove excessive usage of word TEAL in favour of AVM bytecode

neilcampbell commented on docs/tutorials/intro.md:
```diff
@@ -70,102 +72,72 @@ algokit init
+### Using AlgoKit CLI to build and deploy contracts 🛠️
 
-Dappflow is a web-based user interface that let's you visualise accounts, transactions, assets and applications on an Algorand network and also provides ability to deploy and call smart contracts. This works for TestNet, MainNet and also LocalNet. Furthermore, you can also create and fund accounts on LocalNet. While AlgoKit surfaces both a programming interface and a command line interface for in ...
+When using official AlgoKit templates via `algokit init`, in addition to integration with VSCode and JetBrains IDEs, you can manage your projects using the AlgoKit CLI at the convienence of your terminal.
```
```suggestion
When using official AlgoKit templates via `algokit init`, you can manage your projects using the AlgoKit CLI at the convenience of your terminal.
```

neilcampbell commented on docs/tutorials/intro.md:
```diff
@@ -70,102 +72,72 @@ algokit init
+When using official AlgoKit templates via `algokit init`, in addition to integration with VSCode and JetBrains IDEs, you can manage your projects using the AlgoKit CLI at the convienence of your terminal.
 
-Dappflow can be launched from AlgoKit by running the following command from the VS Code terminal.
+For example, to achieve the same results as in the demo earlier, build and deploy the `HelloWorld` contract from the `contract.py` file, you can run the following command from the terminal:
```
```suggestion
For example, to achieve the same results as in the demo earlier, build and deploy the `HelloWorld` contract from the `contract.py` file by running the following command from the terminal:
```

neilcampbell commented on docs/tutorials/intro.md:
```diff
@@ -70,102 +72,72 @@ algokit init
-To issue commands against the LocalNet network you need an account with ALGOs in it. Dappflow lets you easily create one.
-
-Select `Dev Wallets` from the left menu and click on the `Create wallet` button. This will create an account on the LocalNet and fund it with 100 Algos that can be used to test with.
+Which will recompile your python contracts from `contract.py` into TEAL and related contract artifacts.
```
```suggestion
This command will recompile your python contracts from `contract.py` and produce the necessary artefacts for deploying your contract.
```

neilcampbell commented on docs/tutorials/intro.md:
````diff
@@ -70,102 +72,72 @@ algokit init
+```
 
-![AlgoKit Dappflow](../imgs/dappflow6.png)
+This will then execute the deploy script to LocalNet instance and display the same output as in the earlier demo.
````
```suggestion
This will then deploy to your LocalNet instance and display the same output as in the earlier demo.
```

neilcampbell commented on docs/tutorials/intro.md:
```diff
@@ -70,102 +72,72 @@ algokit init
+## Next steps 🚶‍♂️
 
-## Next steps
+We have only covered a tiny fraction of capabilities of the AlgoKit CLI and its related ecosystem of templates and utilities for efficient developer experience.
```
```suggestion
We have only covered a tiny fraction of the capabilities of the AlgoKit CLI and its related ecosystem of templates and utilities for an efficient developer experience.
```

neilcampbell commented on docs/tutorials/intro.md:
```diff
@@ -70,102 +72,72 @@ algokit init
+- To learn more about `Algorand Python`, take a look at the [documentation](https://algorandfoundation.github.io/puya/).
+- To learn more about commands demonstrated in this tutorial, refer to [`init`](../features/init.md) and [`project`](../features/project.md) to get a comprehensive understanding of their further capabilities.
+- If you'd like to structure your `AlgoKit Project` as a monorepo, refer to `workspace` mode as described in [`init`](../features/init.md#workspaces-vs-standalone-projects).
+- If you'd like to **create your own** `AlgoKit` template, refer to the [dedicated tutorials](./algokit-template.md).
```
```suggestion
- If you'd like to **create your own** `AlgoKit` template, refer to the [template tutorial](./algokit-template.md).
```

neilcampbell commented on docs/tutorials/intro.md:
````diff
@@ -70,102 +72,72 @@ algokit init
 
 ```shell
-algokit init -t playground --no-workspace
+algokit init -t python --no-workspace -a preset_name "starter"
````
I'm wondering if we should just use the default and init with a workspace, which would shorten the command to `algokit init -t python -a preset_name "starter"`

aorumbayev replied:
refreshed a screenshot and added a bit more verbose explanation on the structure that user will see with workspace mode by default

aorumbayev commented:
@Loedn I've added a detailed rewrite for the intro tutorial to assume latest python template and the workspace mode (see comments from @neilcampbell above that i've addressed). I've noticed that the videos in header redirect to outdated videos so i've added an explicit note to that section. If any further tweaks would be needed in the docs feel free to open up a new issue or tag us by reopening the existing one (https://github.com/algorandfoundation/algokit-cli/issues/486).
